src/modules/ekf2/EKF2.cpp

```diff
@@ -1430,25 +1406,22 @@ void EKF2::UpdateMagCalibration(const hrt_abstime &timestamp)
 			// Declare all bias estimates invalid if any variances are out of range
 			bool all_estimates_invalid = false;
 
-			float covariances[24];
-			_ekf.covariances_diagonal().copyTo(covariances);
+			const Vector3f mag_bias_variance{_ekf.getMagBiasVariance()};
 
 			for (uint8_t axis_index = 0; axis_index <= 2; axis_index++) {
-				if (covariances[axis_index + 19] < min_var_allowed
-				    || covariances[axis_index + 19] > max_var_allowed) {
+				if (mag_bias_variance(axis_index) < min_var_allowed
+				    || mag_bias_variance(axis_index) > max_var_allowed) {
 					all_estimates_invalid = true;
 				}
 			}
 
 			// Store valid estimates and their associated variances
 			if (!all_estimates_invalid) {
-
-				float states[24];
-				_ekf.getStateAtFusionHorizonAsVector().copyTo(states);
+				const Vector3f mag_bias{_ekf.getMagBias()};
 
 				for (uint8_t axis_index = 0; axis_index <= 2; axis_index++) {
-					_last_valid_mag_cal[axis_index] = states[axis_index + 19];
-					_last_valid_variance[axis_index] = covariances[axis_index + 19];
+					_last_valid_mag_cal[axis_index] = mag_bias(axis_index);
```

Comment:
suggest to make _last_valid_mag_cal & _last_valid_variance of type Vector3f

Reply:
Done.